src/ucs/memory/memtype_cache.c

```diff
@@ -398,3 +402,11 @@ void ucs_memtype_cache_global_cleanup()
         ucs_memtype_cache_destroy(ucs_memtype_cache_global_instance);
     }
 }
+
+UCS_STATIC_INIT {
+    ucs_spinlock_init(&ucs_memtype_cache_global_instance_lock, 0);
+}
+
+UCS_STATIC_CLEANUP {
```

Comment:
can remove ucs_memtype_cache_global_cleanup() and move it's contents to UCS_STATIC_CLEANUP here